.github/workflows/pr.yaml

```diff
@@ -0,0 +1,50 @@
+on:
+  pull_request:
+    types: [opened, synchronize, ready_for_review, edited, closed]
+
+name: "Validate PR"
+jobs:
+  validate:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Pull request title check
+        uses: amannn/action-semantic-pull-request@v4
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v2
+        with:
+          node-version: "14"
+          registry-url: "https://registry.npmjs.org"
+
+      - name: Cache PNPM modules
+        uses: actions/cache@v2
+        with:
+          path: ~/.pnpm-store
+          key: ${{ runner.os }}-${{ hashFiles('**/pnpm-lock.yaml') }}
+          restore-keys: |
+            ${{ runner.os }}-
+
+      - name: Setup PNPM
+        uses: pnpm/action-setup@v2.1.0
+        with:
+          version: 7.0.0
+          run_install: true
+
+      - name: Setup Docker
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Lint & Test
+        run: |
+          pnpm build
+          pnpm lint
+          pnpm build
+          pnpm ci:test-coverage
```

Comment:
Why not simply add these to `ci.yml`? It looks like most of these two worksflow are the same.

Reply:
I split it up because of the pull request title validation, thinking this shouldn't run on push master, but is it possible to put it all together? if is I think there's no problem put all together

Reply:
Yes, you can do it together. Just add an `if` clause to the `Pull request title check` job that checks if it's a PR.

```yml
if github.event_name == 'pull_request'
```

should suffice.

Reply:
Alternatively, keep it as a separate workflow that only runs on PRs!